vendors/espressif/boards/esp32/ports/ble/nimble/iot_ble_hal_gatt_server.c

```diff
@@ -39,155 +39,23 @@
 #include "bt_hal_gatt_server.h"
 #include "iot_ble_hal_internals.h"
 
-#define APP_ID               0
-#define GATT_MAX_ENTITIES    30
+#define APP_ID          0
+#define MAX_SERVICES    20
 
-typedef struct
-{
-    BTGattAttributeType_t type;
-    void * attr;
-    ble_uuid_t * uuid;
-    uint16_t flags;
-    uint16_t parent_handle;
-    uint16_t handle;
-    uint16_t device_handle;
-    uint8_t min_key_size;
-    uint8_t total_attr;
-} BTGattEntity_t;
 
 static struct ble_gatt_access_ctxt g_ctxt;
-
-static BTGattEntity_t xGattTable[ GATT_MAX_ENTITIES ];
-
-static uint8_t xGattTableSize, xAttrNum, xCurrentAttrNum;
-
-static uint16_t xLastSvcHandle;
-
+static struct ble_gatt_svc_def espServices[ MAX_SERVICES + 1 ];
```

Comment:
Why do we need an extra element for espServices?

Reply:
This is required by the Nimble stack. It replaces the length parameters I suppose.